src/System.Console/src/System/ConsolePal.Unix.cs

```diff
@@ -60,7 +60,7 @@ private static SyncTextReader StdInReader
                         () => SyncTextReader.GetSynchronizedTextReader(
                             new StdInStreamReader(
                                 stream: OpenStandardInput(),
-                                encoding: InputEncoding,
+                                encoding: new ConsoleEncoding(Console.InputEncoding), // This ensures no prefix is written to the stream.
```

Comment:
We're not throwing away s_stdInReader when the input encoding is changed... presumably that's by design?